app/models/data_set.rb

```diff
@@ -105,11 +105,22 @@ def appropriate_snac_for_postcode(postcode)
       raise(AmbiguousPostcodeError, response)
     end
 
-    appropriate_authority_for_local_custodian_code(local_custodian_codes.first)&.dig("snac") # there should be 0-1, return nil or first snac
+    authority_hash = appropriate_authority_for_local_custodian_code(local_custodian_codes.first) # there should be 0-1, return first or nil
+    snac_or_gss(authority_hash)
   rescue GdsApi::HTTPNotFound
     nil
   end
 
+  def snac_or_gss(authority_hash)
+    # Not all LAs have SNACs now, so we return nil if the authority hash is nil,
+    # or the SNAC if there is one, or the GSS if there isn't a SNAC
+
+    snac = authority_hash&.dig("snac")
+    return snac if snac.present?
+
+    authority_hash&.dig("gss")
```

Comment:
```
return if authority_hash.blank?

authority_hash["snac"] ? authority_hash["snac"] : authority_hash["gss"]
```
I've refactored the method which I think is a bit clearer. The current code also assumes that we're favouring the snac over the gss. What do you think?